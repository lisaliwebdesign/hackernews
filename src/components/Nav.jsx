import React from "react";
import PropTypes from "prop-types";
import { NavBar } from "../elements/nav";
import { navData } from "../stores/NavStore";

export default class Nav extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: this.props.isLoginPage ? "user-log-in" : "nav-top-t",
      toggle: false,
      isLogInNav: false
    };
  }
  render() {
    const onClick = elementId => {
      this.setState({
        active: elementId,
        isLogInNav: elementId === "user-log-in" ? true : false
      });
    };
    const toggle = () => {
      this.setState({ toggle: !this.state.toggle });
    };
    return (
      <NavBar
        content={navData}
        onClick={onClick}
        active={this.state.active}
        toggleClick={toggle}
        toggle={this.state.toggle}
        isLogInNav={this.props.isLoginPage ? true : this.state.isLogInNav}
      />
    );
  }
}
