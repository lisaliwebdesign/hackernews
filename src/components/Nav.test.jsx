import Nav from "./Nav";

it("<components><Nav - renders component", () => {
  const wrapper = shallow(<Nav isLoginPage={true} />);
  expect(wrapper).toMatchSnapshot();
});
