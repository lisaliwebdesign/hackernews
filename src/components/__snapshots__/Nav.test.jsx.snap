// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<components><Nav - renders component 1`] = `
<NavBar
  active="user-log-in"
  content={
    Array [
      Object {
        "className": "navbar-item",
        "id": "topstories",
        "name": "Top stories",
        "path": "/news/topstories",
      },
      Object {
        "className": "navbar-item",
        "id": "newstories",
        "name": "New stories",
        "path": "/news/newstories",
      },
      Object {
        "className": "navbar-item",
        "id": "beststories",
        "name": "Comments",
        "path": "/news/beststories",
      },
      Object {
        "className": "navbar-item",
        "id": "askstories",
        "name": "Ask",
        "path": "/news/askstories",
      },
      Object {
        "className": "navbar-item",
        "id": "showstories",
        "name": "Show",
        "path": "/news/showstories",
      },
      Object {
        "className": "navbar-item",
        "id": "jobstories",
        "name": "Job",
        "path": "/news/jobstories",
      },
    ]
  }
  isLogInNav={true}
  onClick={[Function]}
  toggle={false}
  toggleClick={[Function]}
/>
`;
